test/e2e/internal/utils/match/request.go

```diff
@@ -23,26 +23,37 @@ import (
 	"github.com/onsi/gomega/gbytes"
 )
 
-// requestHeaderMatcher provides matching for request content.
-// It looks into the debug output of an operation and the match will pass
-// if all the headers exist in every sent request.
-type requestHeaderMatcher []string
+// requestHeaderMatcher provides matching for request headers.
+// Given a url prefix, it looks at the requests sent to the urls which
+// match the given prefix, and check if these requests all contain
+// the specified headers.
+type requestHeaderMatcher struct {
+	urlPrefix string
+	headers   []string
+}
+
+// NewRequestHeaderMatcher returns a request header matcher with
+// the url prefix "".
+func NewRequestHeaderMatcher(headers []string) requestHeaderMatcher {
+	return requestHeaderMatcher{"", headers}
+}
```

Comment:
We don't need this function, `opts.MatchRequestHeaders` can call `NewRequestHeaderMatcherWithPrefix` with `urlPrefix` as empty string.
```suggestion
```

Reply:
removed this function